src/backend/distributed/shared_library_init.c

```diff
@@ -194,6 +195,16 @@ CreateRequiredDirectories(void)
 static void
 RegisterCitusConfigVariables(void)
 {
+	DefineCustomIntVariable(
+		"citus.node_connection_timeout",
+		gettext_noop("Sets the maximum duration to connect to worker nodes."),
+		NULL,
+		&NodeConnectionTimeout,
+		5000, 1000, 60 * 60 * 1000,
```

Comment:
Why is the lower limit this high? A second might very well be too high for a number of cases?